kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourceCheckResult.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.client.CustomResource;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Resource check result
+ * @param passed true if check passed, false otherwise
+ * @param resource null if check passed, else a resource with status updated to reflect problems
```

Comment:
I think we should drop the `passed` parameter and just have a `boolean passed()` method which checks the nullness of `resource`.